lib/queuePopulator/IngestionReader.js

```diff
@@ -190,11 +190,7 @@ class IngestionReader extends LogReader {
             if (err) {
                 return done(err);
             }
-            if (hasMoreLog) {
-                // keep ingesting new log without waiting
-                return this.processLogEntries(params, done);
-            }
-            return done();
+            return done(null, hasMoreLog);
```

Comment:
> The initial bucket scan doesn't respect the cron rule as we continuously list the objects to converge faster, the listing is done in a single _processLogEntries call which makes it ignore the location status (paused, resumed).

I think this is by design, the initial scan should indeed not wait the next cron job... (but it must look for pause & circuit breakers indeed)

and since you moved the checks inside `_processLogEntries()`, it should be fine:
* here we will try to process immediately
* if paused/throttled, it will skip until the "next" cron call

Reply:
ok, actually you moved this "keep ingestion without waiting" from IngestionReader to IngestionPopulator